docs/tutorials/standalone/Soil/layered_soil.jl

```diff
@@ -0,0 +1,171 @@
+af = ArtifactFile(
+    url = "https://caltech.box.com/shared/static/qvbt37xkwz8gveyi6tzzbs0e18trpcsq.csv",
+    filename = "sv_62.csv",
+)
+dataset = ArtifactWrapper(@__DIR__, "sv62", ArtifactFile[af]);
+dataset_path = get_data_folder(dataset);
+data = joinpath(dataset_path, af.filename)
+parameter_data = readdlm(data, ',');
+# Our model treats z as increasing in the upwards direction.
+# Values below the surface are negative.
+# Because of this, we convert the (positive-valued) depth
+# of the data into a monotonically increasing z coordinate value.
+# using a negative sign and the reverse function.
+depth = reverse(-parameter_data[1, :] .* 0.01) # convert to m
+ksat = reverse(parameter_data[6, :] .* 1 / 100.0 / 60.0) # convert cm/min to m/s
+vgα = reverse(parameter_data[4, :] .* 100 * 2) # they report αᵈ; αʷ = 2αᵈ. This experiment is for infiltration (wetting).
+vgn = reverse(parameter_data[5, :])
+residual_frac = reverse(parameter_data[2, :])
+porosity = reverse(parameter_data[3, :]);
+
+# Create fields corresponding to the parameter
+ν = SpaceVaryingInput(depth, porosity, soil_domain.space.subsurface)
+K_sat = SpaceVaryingInput(depth, ksat, soil_domain.space.subsurface)
+θ_r = SpaceVaryingInput(depth, residual_frac, soil_domain.space.subsurface);
+# The specific storativity is not something we have data on, so we
+# approximate it as being constant in depth, and create the parameter field directly:
+S_s = ClimaCore.Fields.zeros(soil_domain.space.subsurface) .+ 1e-3;
+
+# The retention model is a vanGenuchten model with α and n as a function of
+# depth, read from the data:
+hcm = SpaceVaryingInput(
+    depth,
+    (; α = vgα, n = vgn),
+    soil_domain.space.subsurface,
+    vanGenuchten{FT},
+);
+
+# The parameter struct:
+params = ClimaLand.Soil.RichardsParameters(;
+    ν = ν,
+    hydrology_cm = hcm,
+    K_sat = K_sat,
+    S_s = S_s,
+    θ_r = θ_r,
+);
+
+# From here on out, everything should look familiar if you've already gone
+# through the other soil tutorials.
+# Set Boundary conditions:
+# At the top, we use the observed value of Ksat at the top of the domain.
+# Setting the flux to be -Ksat is approximating the top as saturated.
+function top_flux_function(p, t)
+    return -0.0001033
+end
+top_bc = ClimaLand.Soil.FluxBC(top_flux_function)
+bottom_bc = ClimaLand.Soil.FreeDrainage()
+boundary_fluxes = (; top = (water = top_bc,), bottom = (water = bottom_bc,))
+soil = Soil.RichardsModel{FT}(;
+    parameters = params,
+    domain = soil_domain,
+    boundary_conditions = boundary_fluxes,
+    sources = (),
+);
+
+# Initial the state vectors, and set initial conditions
+Y, p, cds = initialize(soil);
+
+# Initial conditions
+Y.soil.ϑ_l .= 0.0353; # read from Figure 4 of Huang et al. 
+
+# We also set the initial conditions of the auxiliary state here:
+set_initial_cache! = make_set_initial_cache(soil)
+set_initial_cache!(p, Y, t0);
+
+# Timestepping:
+stepper = CTS.ARS111()
+err = (FT == Float64) ? 1e-8 : 1e-4
```

Comment:
`Dict`s are not type-stable. Something like `@assert FT in (Float32, Float64)` would be better for this, as it could be optimized out by the compiler.